apps/server/src/modules/fwu-learning-contents/fwu-learning-contents.module.ts

```diff
@@ -0,0 +1,70 @@
+import { Module, NotFoundException } from '@nestjs/common';
+import { S3Client } from '@aws-sdk/client-s3';
+import { AuthorizationModule } from '@src/modules/authorization';
+import { HttpModule } from '@nestjs/axios';
+import { CoreModule } from '@src/core';
+import { ConfigModule } from '@nestjs/config';
+import { Dictionary, IPrimaryKey } from '@mikro-orm/core';
+import { AuthenticationModule } from '@src/modules/authentication/authentication.module';
+import { MikroOrmModule, MikroOrmModuleSyncOptions } from '@mikro-orm/nestjs';
+import { Account, User, Role, School, System, SchoolYear } from '@shared/domain';
+import { createConfigModuleOptions, DB_PASSWORD, DB_URL, DB_USERNAME } from '@src/config';
+import { Logger } from '@src/core/logger';
+import { s3Config, config } from './fwu-learning-contents.config';
+import { FwuLearningContentsController } from './controller/fwu-learning-contents.controller';
+import { FwuLearningContentsUc } from './uc/fwu-learning-contents.uc';
+import { S3Config } from './interface/config';
+
+const providers = [
+	Logger,
+	{
+		provide: 'FWU_S3_Client',
+		useFactory: (configProvider: S3Config) =>
+			new S3Client({
+				region: configProvider.region,
+				credentials: {
+					accessKeyId: configProvider.accessKeyId,
+					secretAccessKey: configProvider.secretAccessKey,
+				},
+				endpoint: configProvider.endpoint,
+				forcePathStyle: true,
+				tls: true,
+			}),
+		inject: ['S3_Config'],
+	},
+	{
+		provide: 'S3_Config',
+		useValue: s3Config,
+	},
+	FwuLearningContentsUc,
+];
+
+const defaultMikroOrmOptions: MikroOrmModuleSyncOptions = {
+	findOneOrFailHandler: (entityName: string, where: Dictionary | IPrimaryKey) =>
+		// eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+		new NotFoundException(`The requested ${entityName}: ${where} has not been found.`),
+};
+
+@Module({
+	imports: [
+		AuthorizationModule,
+		AuthenticationModule,
+		CoreModule,
+		HttpModule,
+		MikroOrmModule.forRoot({
+			...defaultMikroOrmOptions,
+			type: 'mongo',
+			// TODO add mongoose options as mongo options (see database.js)
+			clientUrl: DB_URL,
+			password: DB_PASSWORD,
+			user: DB_USERNAME,
+			entities: [User, Account, Role, School, System, SchoolYear],
+
+			// debug: true, // use it for locally debugging of querys
+		}),
+		ConfigModule.forRoot(createConfigModuleOptions(config)),
+	],
+	controllers: [FwuLearningContentsController],
+	providers,
+})
+export class FwuLearningContentsModule {}
```

Comment:
Im not quite happy with the fact that you put everything into one module

we generally split our modules into api and logic modules, as described here https://hpi-schul-cloud.github.io/schulcloud-server/additional-documentation/nestjs-application/software-architecture.html#modules.

the file-storage module additionally has a root module, since its its own mikroservice, as is the case for you.

I would prefer you to do so as well, but for this it will probably be necessary to split your uc into a uc and a service Looking at your code, that might be non-trivial. Because of your timeconstraints, im willing to let you delay this refactoring until the point where we look at synergies with the file-storage module, as I think that will make it a lot easier and clearer.

Reply:
Yes, I also think it makes sense to discuss the issue when we discuss the synergies between FWU learning contents and file-storage.